lib/mapper/product_mapper.rb

```diff
@@ -19,6 +19,9 @@ def self.map(product, variant, additional_data = {}, is_digital = false, default
                             "width" => default_variant['width'],
                             "depth" => default_variant['depth'],
                             "height" => default_variant['height'],
+                            "releaseDate" => default_variant['releaseDate'],
+                            "datePublished" => default_variant['datePublished'],
+                            "availability" => default_variant['offers'] && default_variant['offers'].select {|x| x['availability']} ? default_variant['offers'].select {|x| x['availability']}[0]['availability'] : ''
```

Comment:
@Jhalter5Stones This will probably need to be modified in response to the feedback I left on the acumen MR.

There is an `Offer` that we use to set the _price_ -- we should pull `availability` from that same offer.

Now, all that said, there are actually a couple concerns here:

I'm fairly confident (though not certain) that availability is (ultimately) something that should be controlled by the WMS (3PL or Deposco -- whichever IP is using).

If availability _is_ managed by the WMS, then updating it in the sync may introduce conflicts or inconsistencies in the data. You may want to implement some logic here. At the very least it's possible we want to do something like

```
If (isPreorder) {
    product.availability =  pre-order;
} else {
    do nothing
}
```

I think we have inventory tracking with BigCommerce/3PL configured for Catholic Answers -- you may want to reach out to Jacob and see if there are any warehouse requirements he knows about that need to be accounted for here.

Reply:
I got specific rules from Chris about availability. If Deposco just handles inventory numbers for products this shouldn't affect that. This only affects availability based on rules for datePublished and releaseDate values based on the information I got from Chris.

Reply:
@Jhalter5Stones We should still be pulling the `availability` value from the same offer object that sets the price (rather than searching for it across all the offers)